src/app/models/query-models/query.ts

```diff
@@ -25,6 +25,8 @@ export class Query {
   private _observationsQuery: {data: string[]};
   // The visual indicator flags the visibility of the query
   private _visible: boolean;
+  // Additional information about the query, e.g. a data table state
+  private _queryBlob: string;
```

Comment:
why is it called queryBlob, not sth. like tableState?

Reply:
also about the type of table state, it would be better to have it structured instead of put everything into a string. The tables states that we need to store: row dimensions, column dimensions, which are readily available in the tableService, as well as the pivot row and pivot column, which we need to think of some way to represent and store.

Reply:
we probably need to rename patientQuery field to subjectQuery, and give it a more appropriate type, e.g. Constraint

Reply:
it's 'queryBlob', because the purpose of this field is to store all additional information about the query, not only the tableState. For subject/observationsQuery it's different, they are stored in separate columns in the database

Reply:
and the query is structured, it is saved as an instance of DataTableState and parsed in the same way

Reply:
I see, still, the queryBlob is a low-level design consideration that does not belong here, we need to store the criteria/constraint/query for each step separately,  and typed, subjectQuery would be of a constraint type, the observationQuery is already structured with {data: string[]}, so that is fine for now, and the tableQuery is of DataTableState type. If the backend needs the blob string, the query class provides a function that assembles all three criteria into a string, and this function will be used by the resource service.